### PR TITLE
Feat/apicalls

### LINKED ISSUE #5: When the server returns a 500 error, the user will see proper error handling
Acceptance Criteria

- [X] Error message if failure on movie list load
- [X] Error message if failure on movie detail load.

### PR DESCRIPTION
**Pull request by brendel-r**
## Describe your changes:
[//]: <> (Which files were changed? What changes were made?)
Added API calls and utilities file. made some changed to our overall structure to be more dynamic!

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have tested the functionality of any new or updated features in the browser, if applicable
- [x] I have added a "Reviewer" to review & merge
- [] If applicable, my code passes all tests

If issue found, describe and refernces user story/projectboard issue

### CODE REVIEW

**Comment by sdtrachsel** [comment]
Looks good! Nice having the fetch and errorhandling implemented.

**Review by sdtrachsel — COMMENTED** [review]
Looks good! Nice having the fetch and errorhandling implemented.

File: src/App.js [hunk 1]
```diff
@@ -2,23 +2,32 @@ import './App.css';
...
+import { fetchApi } from './apiCalls';
+import { formatRatings } from './utilities';
 
 class App extends React.Component {
   constructor() {
     super()
     this.state = {
       movies: [],
-      movieId: ''
+      movieId: null,
```
Comment (niksseif) [inline src/App.js:14]: Assigning 'null' to 'movieID' might create some bugs in your code. The best practice is always to assign the data type that you're expecting to receive from the API. Is 'movieID' an object, array, number, or string?

File: src/components/MovieDetail.js [hunk 2]
```diff
@@ -1,47 +1,62 @@
...
-      backdrop_path: "https://image.tmdb.org/t/p/original//oazPqs1z78LcIOFslbKtJLGlueo.jpg",
-      release_date: "2019-12-04",
-      overview: "Some overview that is full of buzzwords to attempt to entice you to watch this movie! Explosions! Drama! True love! Robots! A cute dog!", average_rating: 6,
-      genres: ["Drama"],
-      budget: 63000000,
-      revenue: 100853753,
-      runtime: 139,
-      tagline: "It's a movie!"
-    }
+      // id: null,
```
Comment (niksseif) [inline src/components/MovieDetail.js:9]: We refer to commented-out code as "dead code." Please ensure it's removed, as it makes your code difficult to read.

File: src/components/MovieDetail.js [hunk 3]
```diff
@@ -1,47 +1,62 @@
...
-  
-  render(){
+    this.props.updateMovieId("");
+  };
+
+  componentDidMount = () => {
+    let newState = { ...this.state };
+    fetchApi(this.props.movie)
+      .then((data) => {
+        newState.movieInfo = formatMovieInfo(data.movie);
```
Comment (niksseif) [inline src/components/MovieDetail.js:34]: What is 'newState'? Make sure that your naming is descriptive.

**Pull request merged**